src/io/flutter/pub/PubRoot.java

```diff
@@ -292,13 +292,48 @@ public VirtualFile getLib() {
   }
 
   /**
+   * Returns a file in lib if it exists.
+   */
+  @Nullable
+  public VirtualFile getFileToOpen() {
+    VirtualFile main = getLibMain();
+    if (main != null) {
+      return main;
+    }
+    if (lib != null) {
+      VirtualFile[] files = lib.getChildren();
```

Comment:
And here (and below)?  (Or are we not following that convention anymore?)

Reply:
I'll send you a new PR with 'final' added.